src/components/FileBrowser.vue

```diff
@@ -0,0 +1,279 @@
+<script>
+import { sizeFormatter } from '../utils/mixins';
+
+const GIRDER_FOLDER_ENDPOINT = 'folder';
+const GIRDER_ITEM_ENDOINT = 'item';
+const ICON_MAP = {
+  folder: 'folder',
+  item: 'file',
+};
+
+export default {
+  props: {
+    // location should have properties `type`, `id`
+    location: {
+      type: Object,
+      required: true,
+    },
+    selectEnabled: {
+      type: Boolean,
+      default: true,
+    },
+    multiSelectEnabled: {
+      type: Boolean,
+      default: false,
+    },
+    uploadEnabled: {
+      type: Boolean,
+      default: false,
+    },
+    newItemEnabled: {
+      type: Boolean,
+      default: true,
+    },
+    newFolderEnabled: {
+      type: Boolean,
+      default: true,
+    },
+  },
+  mixins: [sizeFormatter],
+  inject: ['girderRest'],
+  data() {
+    return {
+      rowsLoading: false,
+      breadcrumbLoading: false,
+      pagination: {
+        rowsPerPage: 10,
+        page: 1,
+      },
+      selected: [],
+      refreshCounter_: 0, // https://github.com/girder/girder_web_components/issues/39
+    };
+  },
+  computed: {
+    loading() { return this.rowsLoading || this.breadcrumbLoading; },
+    totalItems() { return this.counts.nFolders + this.counts.nItems; },
+  },
+  asyncComputed: {
+    breadcrumb: {
+      default: { root: {}, path: [] },
+      async get() {
+        this.breadcrumbLoading = true;
+        const breadcrumb = { root: {}, path: [] };
+        let { id, type } = this.location;
+        while (type) {
+          const { data } = (await this.girderRest.get(`${type}/${id}`));
+          const entity = {
+            name: data._modelType !== 'user' ? data.name : data.login,
+            id: data._id,
+            type: data._modelType,
+            parentId: data.parentId,
+            parentType: data.parentCollection,
+          };
+          if (entity.type === 'folder') {
+            breadcrumb.path.unshift(entity);
+            id = entity.parentId;
+            type = entity.parentType;
+          } else {
+            breadcrumb.root = entity;
+            break;
+          }
+        }
+        this.breadcrumbLoading = false;
+        return breadcrumb;
+      },
+    },
+    counts: {
+      default: { nFolders: 0, nItems: 0 },
+      async get() {
+        const endpoint = `${this.location.type}/${this.location.id}/details`;
+        const { data } = (await this.girderRest.get(endpoint));
+        return {
+          nFolders: data.nFolders ? data.nFolders : 0,
+          nItems: data.nItems ? data.nItems : 0,
+        };
+      },
+      watch() {
+        return [
+          this.pagination,
+          this.location,
+        ];
+      },
+    },
+    rows: {
```

Comment:
@subdavis Thanks for updating this rows() and the comment on limit calculation. @jbeezley @zachmullen It seems the doubled fetch still exists and its caused by vue-async-computed. The rows() depends on counts(), and it's being evaluated once before counts() updates and once after counts() updates. I created a [jsfiddle ](https://jsfiddle.net/jfmorio/fpyowz6v/1/) to illustrate the probelm. I have been using vue-async-computed in other projects and I like it, but this seems is a case where normal data() + watch:{} is better suited.